benchmarks/swe_bench/swe_bench.py

```diff
@@ -0,0 +1,372 @@
+"""SWE-bench: Can Language Models Resolve Real-World GitHub Issues?
+
+Carlos E. Jimenez, John Yang, Alexander Wettig, Shunyu Yao, Kexin Pei, Ofir Press, Karthik Narasimhan
+https://arxiv.org/abs/2310.06770
+"""
+
+import json
+import os
+import re
+import shlex
+from pathlib import Path
+from typing import Callable
+
+from docker import DockerClient
+from swebench.harness.constants import (
+    APPLY_PATCH_FAIL,
+    MAP_REPO_TO_INSTALL,
+    MAP_REPO_VERSION_TO_SPECS,
+    RESET_FAILED,
+    TESTS_ERROR,
+    TESTS_TIMEOUT,
+)
+from swebench.harness.log_parsers import MAP_REPO_TO_PARSER
+from swebench.harness.utils import get_test_directives
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import FieldSpec, Sample, hf_dataset
+from inspect_ai.scorer import Score, Scorer, Target, mean, scorer, std
+from inspect_ai.solver import (
+    Plan,
+    TaskState,
+    generate,
+    system_message,
+    use_tools,
+)
+from inspect_ai.tool import bash
+from inspect_ai.util import sandbox
+
+INPUT_PROMPT = "Please solve the following issue:\n\n{issue_text}"
+COMPOSE_FILE_DIR = Path(__file__).parent / "resources/compose_files/"
+os.makedirs(COMPOSE_FILE_DIR, exist_ok=True)
+
+SAMPLE_TO_IMAGE_PATH = COMPOSE_FILE_DIR / "sample_to_image.json"
+
+DEFAULT_PLAN = Plan(
+    [
+        system_message("Please solve the task below."),
+        use_tools([bash()]),
+        generate(),
+    ],
+    name="simple_bash_plan",
+)
+
+DEFAULT_MAX_MESSAGES = 10
+
+
+@task
+def swe_bench(
+    dataset: str = "princeton-nlp/SWE-bench_Verified",
+    split: str = "test",
+    plan: Plan = DEFAULT_PLAN,
+    max_messages: int = DEFAULT_MAX_MESSAGES,
+    filter: Callable[[Sample], bool] | None = None,
```

Comment:
I we can figure out a way to make `filter` serialisable then this `@task` function will support retries when run from the CLI e.g. `inspect eval swe_bench.py` will offer `inspect eval-retry swe_bench.py` on a failure. We obviously already get retries from `eval_set()` but it would be nice to have it from the CLI as well.

What I'm thinking is that if there are _common_ filter conditions (e.g. by id or some subcategory) we could make those explicit parameters. Then, we could also keep `filter` as a failsafe for more flexibility.

Note that later this week I am hoping to add support for `--plan` from the CLI (where the plan can live in another source file or even in a package) which will make this CLI w/ retries scenario more salient.

Reply:
Is this something you want to do as part of this PR? Or just thinking out loud?

Reply:
We can take a pass at this after the PR is merged.